src/dealDashboard/dealDashboard.ts

```diff
@@ -1,279 +1,110 @@
-  //           this.hydrateUserData();
-  //         }
-  //       });
-  //   }
-  // }
+  async activate(_, __, navigationInstruction) {
+    this.setThreadIdFromRoute(navigationInstruction);
+    this.routeChangeEvent = this.eventAggregator.subscribe("router:navigation:complete", (response) => {
+      this.setThreadIdFromRoute(response.instruction);
+    });
+
+    this.dealId = navigationInstruction.params.address;
+    await this.dealService.ensureInitialized();
+    this.deal = this.dealService.deals.get(this.dealId);
+    await this.deal.ensureInitialized();
+    this.clauses = this.deal.registrationData.terms.clauses;
+  }
 
-  // async claim(): Promise<void> {
-  //   if (await this.validateClosedOrPaused()) {
-  //     return;
-  //   }
+  deactivate() {
+    this.routeChangeEvent.dispose();
+  }
 
-  //   if (this.deal.claimingIsOpen && this.deal.userCanClaim) {
-  //     if (!this.dealTokenToReceive?.gt(0)) {
-  //       this.eventAggregator.publish("handleValidationError", `Please enter the amount of ${this.deal.dealTokenInfo.symbol} you wish to receive`);
-  //     } else if (this.deal.userClaimableAmount.lt(this.dealTokenToReceive)) {
-  //       this.eventAggregator.publish("handleValidationError", `The amount of ${this.deal.dealTokenInfo.symbol} you are requesting exceeds your claimable amount`);
-  //     } else {
-  //       this.deal.claim(this.dealTokenToReceive);
-  //     }
-  //   }
-  // }
+  accountAddressChanged(newAddress: Address){
+    this.connected = this.deal && ([
+      this.deal.registrationData.proposalLead,
+      ...this.deal.registrationData.primaryDAO.representatives.map(representative => representative.address),
+      ...this.deal.registrationData.partnerDAO.representatives.map(representative => representative.address),
+    ].includes(newAddress));
+  }
 
-  // async retrieve(): Promise<void> {
-  //   if (await this.validateClosedOrPaused()) {
-  //     return;
-  //   }
+  private setThreadIdFromRoute(navigationInstruction): void {
+    const currentRoute = navigationInstruction.params.childRoute;
+    if (currentRoute && currentRoute.includes("/")) {
+      this.activeClause = navigationInstruction.params.childRoute.split("/")[1];
+    } else {
+      this.activeClause = undefined;
+    }
+  }
 
-  //   if (this.deal.userCanRetrieve) {
-  //     this.deal.retrieveFundingTokens()
-  //       .then((receipt) => {
-  //         if (receipt) {
-  //           this.hydrateUserData();
-  //         }
-  //       });
-  //   }
-  // }
+  /**
+   * Adds a new discussion thread to the deal
+   * (Currently saves the thread to the local storage- this should be replaced with a data-storage call)
+   *
+   * @param topic the discussion topic
+   * @param id the id of the clause the discussion is for or null if it is a general discussion
+   */
+  private addOrReadDiscussion = async (topic: string, discussionHash: string, clauseHash: string | null, clauseIndex: number | null): Promise<void> => {
```

Comment:
for simplicity and consistency, define functions using normal typescript syntax.